Note: The failure only occurs on `cli` tests `netconfig` works, so watch out for that when testing

If I apply this patch on top of `devel` I get the following failure

```
TASK [junos_netconf : Ensure netconf is enabled] *************************************************************************************************************************************************************************************************
task path: /home/johnb/git/ansible-inc/ansible/test/integration/targets/junos_netconf/tests/cli/netconf.yaml:5
 14862 1486376331.79497: sending task start callback
 14862 1486376331.79502: entering _queue_task() for vsrx01/junos_netconf
 14862 1486376331.79506: Creating lock for junos_netconf
 14862 1486376331.79694: worker is 1 (out of 1 available)
 14862 1486376331.79735: exiting _queue_task() for vsrx01/junos_netconf
 14862 1486376331.79800: done queuing things up, now waiting for results queue to drain
 14862 1486376331.79807: waiting for pending results...
 14918 1486376331.80844: running TaskExecutor() for vsrx01/TASK: junos_netconf : Ensure netconf is enabled
 14918 1486376331.81273: in run()
 14918 1486376331.81443: calling self._execute()
 14918 1486376331.82590: Loading FilterModule 'core' from /home/johnb/git/ansible-inc/ansible/lib/ansible/plugins/filter/core.py
 14918 1486376331.86287: Loading FilterModule 'ipaddr' from /home/johnb/git/ansible-inc/ansible/lib/ansible/plugins/filter/ipaddr.py
 14918 1486376331.86390: Loading FilterModule 'json_query' from /home/johnb/git/ansible-inc/ansible/lib/ansible/plugins/filter/json_query.py
 14918 1486376331.86416: Loading FilterModule 'mathstuff' from /home/johnb/git/ansible-inc/ansible/lib/ansible/plugins/filter/mathstuff.py
 14918 1486376331.86512: Loading TestModule 'core' from /home/johnb/git/ansible-inc/ansible/lib/ansible/plugins/test/core.py
 14918 1486376331.86532: Loading TestModule 'files' from /home/johnb/git/ansible-inc/ansible/lib/ansible/plugins/test/files.py
 14918 1486376331.86548: Loading TestModule 'mathstuff' from /home/johnb/git/ansible-inc/ansible/lib/ansible/plugins/test/mathstuff.py
 14918 1486376331.89594: Loading FilterModule 'core' from /home/johnb/git/ansible-inc/ansible/lib/ansible/plugins/filter/core.py (found_in_cache=True, class_only=False)
 14918 1486376331.89610: Loading FilterModule 'ipaddr' from /home/johnb/git/ansible-inc/ansible/lib/ansible/plugins/filter/ipaddr.py (found_in_cache=True, class_only=False)
 14918 1486376331.89619: Loading FilterModule 'json_query' from /home/johnb/git/ansible-inc/ansible/lib/ansible/plugins/filter/json_query.py (found_in_cache=True, class_only=False)
 14918 1486376331.89627: Loading FilterModule 'mathstuff' from /home/johnb/git/ansible-inc/ansible/lib/ansible/plugins/filter/mathstuff.py (found_in_cache=True, class_only=False)
 14918 1486376331.89659: Loading TestModule 'core' from /home/johnb/git/ansible-inc/ansible/lib/ansible/plugins/test/core.py (found_in_cache=True, class_only=False)
 14918 1486376331.89668: Loading TestModule 'files' from /home/johnb/git/ansible-inc/ansible/lib/ansible/plugins/test/files.py (found_in_cache=True, class_only=False)
 14918 1486376331.89677: Loading TestModule 'mathstuff' from /home/johnb/git/ansible-inc/ansible/lib/ansible/plugins/test/mathstuff.py (found_in_cache=True, class_only=False)
 14918 1486376331.90281: Loading Connection 'local' from /home/johnb/git/ansible-inc/ansible/lib/ansible/plugins/connection/local.py
 14918 1486376331.90604: Loading ShellModule 'csh' from /home/johnb/git/ansible-inc/ansible/lib/ansible/plugins/shell/csh.py
 14918 1486376331.90782: Loading ShellModule 'fish' from /home/johnb/git/ansible-inc/ansible/lib/ansible/plugins/shell/fish.py
 14918 1486376331.91104: Loading ShellModule 'powershell' from /home/johnb/git/ansible-inc/ansible/lib/ansible/plugins/shell/powershell.py
 14918 1486376331.91209: Loading ShellModule 'sh' from /home/johnb/git/ansible-inc/ansible/lib/ansible/plugins/shell/sh.py
 14918 1486376331.91322: Loading ShellModule 'sh' from /home/johnb/git/ansible-inc/ansible/lib/ansible/plugins/shell/sh.py (found_in_cache=True, class_only=False)
 14918 1486376331.91388: in VariableManager get_vars()
 14918 1486376331.91623: done with get_vars()
 14918 1486376331.91698: Loading Connection 'local' from /home/johnb/git/ansible-inc/ansible/lib/ansible/plugins/connection/local.py (found_in_cache=True, class_only=True)
 14918 1486376331.91810: Loading ActionModule 'normal' from /home/johnb/git/ansible-inc/ansible/lib/ansible/plugins/action/normal.py
 14918 1486376331.91829: starting attempt loop
 14918 1486376331.91836: running the handler
 14918 1486376331.91938: ANSIBALLZ: Using lock for junos_netconf
 14918 1486376331.91967: ANSIBALLZ: Acquiring lock
 14918 1486376331.91999: ANSIBALLZ: Lock acquired: 140519988834640
 14918 1486376331.92034: ANSIBALLZ: Creating module
 14918 1486376332.10443: ANSIBALLZ: Writing module
 14918 1486376332.10490: ANSIBALLZ: Renaming module
 14918 1486376332.10501: ANSIBALLZ: Done creating module
Using module file /home/johnb/git/ansible-inc/ansible/lib/ansible/modules/network/junos/junos_netconf.py
 14918 1486376332.10594: _low_level_execute_command(): starting
 14918 1486376332.10601: _low_level_execute_command(): executing: /bin/sh -c 'echo ~ && sleep 0'
<vsrx01> ESTABLISH LOCAL CONNECTION FOR USER: johnb
 14918 1486376332.10616: in local.exec_command()
<vsrx01> EXEC /bin/sh -c 'echo ~ && sleep 0'
 14918 1486376332.10622: opening command with Popen()
 14918 1486376332.11293: done running command with Popen()
 14918 1486376332.11309: getting output with communicate()
 14918 1486376332.11421: done communicating
 14918 1486376332.11428: done with local.exec_command()
 14918 1486376332.11440: _low_level_execute_command() done: rc=0, stdout=/home/johnb
, stderr=
 14918 1486376332.11455: _low_level_execute_command(): starting
 14918 1486376332.11465: _low_level_execute_command(): executing: /bin/sh -c '( umask 77 && mkdir -p "` echo /home/johnb/.ansible/tmp/ansible-tmp-1486376332.11-163546815939294 `" && echo ansible-tmp-1486376332.11-163546815939294="` echo /home/johnb/.ansible/tmp/ansible-tmp-1486376332.11-163546815939294 `" ) && sleep 0'
 14918 1486376332.11473: in local.exec_command()
<vsrx01> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/johnb/.ansible/tmp/ansible-tmp-1486376332.11-163546815939294 `" && echo ansible-tmp-1486376332.11-163546815939294="` echo /home/johnb/.ansible/tmp/ansible-tmp-1486376332.11-163546815939294 `" ) && sleep 0'
 14918 1486376332.11481: opening command with Popen()
 14918 1486376332.11882: done running command with Popen()
 14918 1486376332.11897: getting output with communicate()
 14918 1486376332.12786: done communicating
 14918 1486376332.12794: done with local.exec_command()
 14918 1486376332.12806: _low_level_execute_command() done: rc=0, stdout=ansible-tmp-1486376332.11-163546815939294=/home/johnb/.ansible/tmp/ansible-tmp-1486376332.11-163546815939294
, stderr=
 14918 1486376332.12819: transferring module to remote /home/johnb/.ansible/tmp/ansible-tmp-1486376332.11-163546815939294/junos_netconf.py
<vsrx01> PUT /tmp/tmpKvlV_I TO /home/johnb/.ansible/tmp/ansible-tmp-1486376332.11-163546815939294/junos_netconf.py
 14918 1486376332.13046: done transferring module to remote
 14918 1486376332.13065: _low_level_execute_command(): starting
 14918 1486376332.13073: _low_level_execute_command(): executing: /bin/sh -c 'chmod u+x /home/johnb/.ansible/tmp/ansible-tmp-1486376332.11-163546815939294/ /home/johnb/.ansible/tmp/ansible-tmp-1486376332.11-163546815939294/junos_netconf.py && sleep 0'
 14918 1486376332.13080: in local.exec_command()
<vsrx01> EXEC /bin/sh -c 'chmod u+x /home/johnb/.ansible/tmp/ansible-tmp-1486376332.11-163546815939294/ /home/johnb/.ansible/tmp/ansible-tmp-1486376332.11-163546815939294/junos_netconf.py && sleep 0'
 14918 1486376332.13086: opening command with Popen()
 14918 1486376332.13752: done running command with Popen()
 14918 1486376332.13770: getting output with communicate()
 14918 1486376332.13789: done communicating
 14918 1486376332.13795: done with local.exec_command()
 14918 1486376332.13817: _low_level_execute_command() done: rc=0, stdout=, stderr=
 14918 1486376332.13858: _low_level_execute_command(): starting
 14918 1486376332.13892: _low_level_execute_command(): executing: /bin/sh -c 'python /home/johnb/.ansible/tmp/ansible-tmp-1486376332.11-163546815939294/junos_netconf.py; rm -rf "/home/johnb/.ansible/tmp/ansible-tmp-1486376332.11-163546815939294/" > /dev/null 2>&1 && sleep 0'
 14918 1486376332.13921: in local.exec_command()
<vsrx01> EXEC /bin/sh -c 'python /home/johnb/.ansible/tmp/ansible-tmp-1486376332.11-163546815939294/junos_netconf.py; rm -rf "/home/johnb/.ansible/tmp/ansible-tmp-1486376332.11-163546815939294/" > /dev/null 2>&1 && sleep 0'
 14918 1486376332.13965: opening command with Popen()
 14918 1486376332.14662: done running command with Popen()
 14918 1486376332.14713: getting output with communicate()
 14918 1486376346.10420: done communicating
 14918 1486376346.10427: done with local.exec_command()
 14918 1486376346.10439: _low_level_execute_command() done: rc=0, stdout=, stderr=Traceback (most recent call last):
  File "/tmp/ansible_oMTwKP/ansible_module_junos_netconf.py", line 152, in <module>
    main()
  File "/tmp/ansible_oMTwKP/ansible_module_junos_netconf.py", line 119, in main
    supports_check_mode=True)
  File "/tmp/ansible_oMTwKP/ansible_modlib.zip/ansible/module_utils/network.py", line 134, in __init__
  File "/tmp/ansible_oMTwKP/ansible_modlib.zip/ansible/module_utils/network.py", line 170, in connect
  File "/tmp/ansible_oMTwKP/ansible_modlib.zip/ansible/module_utils/junos.py", line 291, in connect
  File "/tmp/ansible_oMTwKP/ansible_modlib.zip/ansible/module_utils/shell.py", line 237, in connect
  File "/tmp/ansible_oMTwKP/ansible_modlib.zip/ansible/module_utils/shell.py", line 117, in open
  File "/tmp/ansible_oMTwKP/ansible_modlib.zip/ansible/module_utils/shell.py", line 133, in receive
  File "/home/johnb/.local/lib/python2.7/site-packages/paramiko/channel.py", line 667, in recv
    raise socket.timeout()
socket.timeout

 14918 1486376346.10461: done with _execute_module (junos_netconf, {'_ansible_version': '2.3.0', '_ansible_selinux_special_fs': [u'fuse', u'nfs', u'vboxsf', u'ramfs'], '_ansible_no_log': False, '_ansible_module_name': u'junos_netconf', u'provider': {u'username': u'root', u'host': u'vsrx01', u'password': u'ans123', u'transport': u'cli'}, '_ansible_verbosity': 5, '_ansible_syslog_facility': u'LOG_USER', u'state': u'present', '_ansible_diff': False, '_ansible_debug': True, '_ansible_check_mode': False})
 14918 1486376346.10473: handler run complete
 14918 1486376346.10498: attempt loop complete, returning result
 14918 1486376346.10506: _execute() done
 14918 1486376346.10509: dumping result to json
 14918 1486376346.10515: done dumping result, returning
 14918 1486376346.10526: done running TaskExecutor() for vsrx01/TASK: junos_netconf : Ensure netconf is enabled
 14918 1486376346.10530: sending task result
 14918 1486376346.10565: done sending task result
 14918 1486376346.10569: WORKER PROCESS EXITING
 14862 1486376346.10637: in VariableManager get_vars()
 14862 1486376346.10721: done with get_vars()
 14862 1486376346.10745: marking vsrx01 as failed
 14862 1486376346.10754: marking host vsrx01 failed, current state: HOST STATE: block=4, task=1, rescue=0, always=0, run_state=ITERATING_TASKS, fail_state=FAILED_NONE, pending_setup=False, tasks child state? (HOST STATE: block=0, task=4, rescue=0, always=0, run_state=ITERATING_TASKS, fail_state=FAILED_NONE, pending_setup=False, tasks child state? (HOST STATE: block=0, task=2, rescue=0, always=0, run_state=ITERATING_TASKS, fail_state=FAILED_NONE, pending_setup=False, tasks child state? (None), rescue child state? (None), always child state? (None), did rescue? False, did start at task? False), rescue child state? (None), always child state? (None), did rescue? False, did start at task? False), rescue child state? (None), always child state? (None), did rescue? False, did start at task? False
 14862 1486376346.10761: ^ failed state is now: HOST STATE: block=4, task=1, rescue=0, always=0, run_state=ITERATING_TASKS, fail_state=FAILED_NONE, pending_setup=False, tasks child state? (HOST STATE: block=0, task=4, rescue=0, always=0, run_state=ITERATING_TASKS, fail_state=FAILED_NONE, pending_setup=False, tasks child state? (HOST STATE: block=0, task=2, rescue=0, always=0, run_state=ITERATING_COMPLETE, fail_state=FAILED_TASKS, pending_setup=False, tasks child state? (None), rescue child state? (None), always child state? (None), did rescue? False, did start at task? False), rescue child state? (None), always child state? (None), did rescue? False, did start at task? False), rescue child state? (None), always child state? (None), did rescue? False, did start at task? False
 14862 1486376346.10766: getting the next task for host vsrx01
 14862 1486376346.10773: done getting next task for host vsrx01
 14862 1486376346.10775:  ^ task is: None
 14862 1486376346.10778:  ^ state is: HOST STATE: block=4, task=1, rescue=0, always=0, run_state=ITERATING_COMPLETE, fail_state=FAILED_TASKS, pending_setup=False, tasks child state? (None), rescue child state? (None), always child state? (None), did rescue? False, did start at task? False
An exception occurred during task execution. The full traceback is:
Traceback (most recent call last):
  File "/tmp/ansible_oMTwKP/ansible_module_junos_netconf.py", line 152, in <module>
    main()
  File "/tmp/ansible_oMTwKP/ansible_module_junos_netconf.py", line 119, in main
    supports_check_mode=True)
  File "/tmp/ansible_oMTwKP/ansible_modlib.zip/ansible/module_utils/network.py", line 134, in __init__
  File "/tmp/ansible_oMTwKP/ansible_modlib.zip/ansible/module_utils/network.py", line 170, in connect
  File "/tmp/ansible_oMTwKP/ansible_modlib.zip/ansible/module_utils/junos.py", line 291, in connect
  File "/tmp/ansible_oMTwKP/ansible_modlib.zip/ansible/module_utils/shell.py", line 237, in connect
  File "/tmp/ansible_oMTwKP/ansible_modlib.zip/ansible/module_utils/shell.py", line 117, in open
  File "/tmp/ansible_oMTwKP/ansible_modlib.zip/ansible/module_utils/shell.py", line 133, in receive
  File "/home/johnb/.local/lib/python2.7/site-packages/paramiko/channel.py", line 667, in recv
    raise socket.timeout()
socket.timeout

fatal: [vsrx01]: FAILED! => {
    "changed": false, 
    "failed": true, 
    "invocation": {
        "module_name": "junos_netconf"
    }, 
    "module_stderr": "Traceback (most recent call last):\n  File \"/tmp/ansible_oMTwKP/ansible_module_junos_netconf.py\", line 152, in <module>\n    main()\n  File \"/tmp/ansible_oMTwKP/ansible_module_junos_netconf.py\", line 119, in main\n    supports_check_mode=True)\n  File \"/tmp/ansible_oMTwKP/ansible_modlib.zip/ansible/module_utils/network.py\", line 134, in __init__\n  File \"/tmp/ansible_oMTwKP/ansible_modlib.zip/ansible/module_utils/network.py\", line 170, in connect\n  File \"/tmp/ansible_oMTwKP/ansible_modlib.zip/ansible/module_utils/junos.py\", line 291, in connect\n  File \"/tmp/ansible_oMTwKP/ansible_modlib.zip/ansible/module_utils/shell.py\", line 237, in connect\n  File \"/tmp/ansible_oMTwKP/ansible_modlib.zip/ansible/module_utils/shell.py\", line 117, in open\n  File \"/tmp/ansible_oMTwKP/ansible_modlib.zip/ansible/module_utils/shell.py\", line 133, in receive\n  File \"/home/johnb/.local/lib/python2.7/site-packages/paramiko/channel.py\", line 667, in recv\n    raise socket.timeout()\nsocket.timeout\n", 
    "module_stdout": "", 
    "msg": "MODULE FAILURE", 
    "rc": 0
}
 14862 1486376346.10819: no more pending results, returning what we have

```

--- a/lib/ansible/plugins/terminal/junos.py
+++ b/lib/ansible/plugins/terminal/junos.py
@@ -26,11 +26,17 @@ from ansible.plugins.terminal import TerminalBase
 from ansible.errors import AnsibleConnectionFailure
 
 
+try:
+    from __main__ import display
+except ImportError:
+    from ansible.utils.display import Display
+    display = Display()
+
 class TerminalModule(TerminalBase):
 
     terminal_prompts_re = [
         re.compile(r"[\r\n]?[\w+\-\.:\/\[\]]+(?:\([^\)]+\)){,3}(?:>|#) ?$"),
-        re.compile(r"[\r\n]?[\w+\-\.:\/\[\]]+(?:\([^\)]+\)){,3}(?:>|#) ?$"),
+        re.compile(r"[\r\n]?[\w+\-\.:\/\[\]]+(?:\([^\)]+\)){,3}(?:>|%) ?$"),
     ]
 
     terminal_errors_re = [
@@ -40,6 +46,10 @@ class TerminalModule(TerminalBase):
 
     def on_open_shell(self):
         try:
+            prompt = self._get_prompt()
+            if prompt.strip().endswith('%'):
+                display.vvv('starting cli', self._connection._play_context.remote_addr)
+                self._exec_cli_command('cli')
             for c in ['set cli timestamp disable', 'set cli screen-length 0']:
                 self._exec_cli_command(c)
         except AnsibleConnectionFailure:

